### PR TITLE
Podcast (aka Show) support

### PR DESCRIPTION
**Pull request by gordonshieh**

### CODE REVIEW

**Comment by jpochyla** [comment]
Whoo, amazing, thanks @gordonshieh! Do you also plan on working on playback? It'd be definitely more involved, so if you prefer, I can try to hack that in sometime next week.

**Comment by gordonshieh** [comment]
If it's like `Track`, I might be able to figure out the playback for podcasts

**Comment by jpochyla** [comment]
Incomplete list of things I think we need:

- `psst-core`
  - Implement `metadata::ToMediaPath` for `psst_protocol::metadata::Episode`.
  - Add caching for episodes, maybe by generalising the `Cache` a bit?.
  - Fill the `Podcast` branch in `player::item::load_media_path`.
  - For "continue listening to episode" feature:
    - Support starting playback from a specific position.
    - Nice to have: Rewrite stream storage as block-based, with a persisted bitmap of downloaded blocks, so we can cache even incomplete track/episode content.
- `psst-gui`
  - Add data type for episodes (the PR has `Arc<Track>` placeholder for it now).
  - Generalise or duplicate `PlaybackController` so we can play episodes.
  - Add search for shows/episodes.
  - For "continue listening to episode" feature:
    - Persist currently playing track/episode to config.

The `core` part of stuff is slightly overcomplicated now, I plan to revisit it and simplify when adding local track playback.

**Comment by jpochyla** [comment]
Basic episode playback is implemented, needs a BIG cleanup, the code is getting absolutely ridiculous 🙄

**Comment by jpochyla** [comment]
It's a bit better now.

**Comment by gordonshieh** [comment]
You've got playback working much faster than I ever can! 😅 
Pushed a small commit to get it compiling successfully again

File: psst-gui/Cargo.toml [hunk 1]
```diff
@@ -11,26 +11,27 @@ repository = "https://github.com/jpochyla/psst"
...
 log = "0.4"
 lru-cache = "0.1"
-once_cell = "1.8"
+once_cell = "1.9"
 open = "2.0"
 parking_lot = "0.11"
 platform-dirs = "0.3"
 rand = "0.8"
-raw-window-handle = "0.3"
+raw-window-handle = "0.4"
```
Comment (gordonshieh) [inline psst-gui/Cargo.toml:27]: this version bump needs to be reflected on the druid fork too. having issues building on Windows.

```
error[E0599]: the method `raw_window_handle` exists for reference `&WindowHandle`, but its trait bounds were not satisfied
   --> psst-gui\src\controller\playback.rs:132:43
    |
132 |                 let handle = match window.raw_window_handle() {
    |                                           ^^^^^^^^^^^^^^^^^ method cannot be called on `&WindowHandle` due to unsatisfied trait bounds
    |
   ::: C:\Users\gordo\.cargo\git\checkouts\druid-64a54200c8cd3ebf\625374b\druid-shell\src\window.rs:173:1
    |
173 | pub struct WindowHandle(pub(crate) backend::WindowHandle);
    | ---------------------------------------------------------- doesn't satisfy `WindowHandle: HasRawWindowHandle`
    |
    = note: the following trait bounds were not satisfied:
            `WindowHandle: HasRawWindowHandle`
            which is required by `&WindowHandle: HasRawWindowHandle`
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   | use raw_window_handle::HasRawWindowHandle;
    |
```

Comment (jpochyla) [reply psst-gui/Cargo.toml:27]: Ooops, let's revert it then.

**Comment by gordonshieh** [comment]
rebased and ready for review! i'll try for playback persist in config in another PR

**Comment by jpochyla** [comment]
Yea let's merge this already. Let's do proper episode detail and persistent playback in another PRs.

**Pull request merged**